vendor/k8s.io/kubernetes/staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -248,6 +248,11 @@ func (j *JSONPath) evalArray(input []reflect.Value, node *ArrayNode) ([]reflect.
 		}
 		if params[0].Value < 0 {
 			params[0].Value += value.Len()
+
+			// if value still less than zero, error out
+			if params[0].Value < 0 {
```

Comment:
I was unable to find docs on how to handle this properly in jsonpath. In Python, I am able to wrap around once before I receive an out of bounds error:

```
s = [1,2,3,4]
s[-1] # 4
s[-5] # list index out of range
```

I have updated the patch to continue to wrap around until the specified index is > 0 for `params[0].Value` and `params[1].Value`, but I'm not still sure if I am handling this correctly